nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -239,6 +248,17 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
 
     }
 
+    protected void writeWorkflowInitiator(JsonGenerator jg, String workflowInitiator) throws IOException {
+        if (ctx.getFetched(ENTITY_TYPE).contains(FETCH_WORKFLOW_INITATIOR)) {
+            NuxeoPrincipal principal = userManager.getPrincipal(workflowInitiator);
+            if (principal != null) {
+                writeEntityField("workflowInitiator", principal, jg);
+                return;
+            }
+        }
+        jg.writeStringField("workflowInitiator", workflowInitiator);
```

Comment:
Isn't weird to have workflowInitiator whereas we don't have the appropriate header ?

Reply:
Nope, we always want the workflow initiator, the `workflowInitiator` fetch property allowing to have the whole `Principal` object instead of just the id.